src/main/java/pl/sdacademy/booking/validator/NewEventDtoValidator.java

```diff
@@ -1,63 +1,61 @@
 package pl.sdacademy.booking.validator;
 
-import org.hibernate.grammars.hql.HqlParser;
+import org.apache.commons.lang3.StringUtils;
 import pl.sdacademy.booking.model.NewEventDto;
+import pl.sdacademy.booking.util.TimeNow;
 
 import java.time.Duration;
 import java.time.LocalDateTime;
 import java.util.ArrayList;
 import java.util.List;
-import java.util.Locale;
 
 public class NewEventDtoValidator {
-    //najcześciej zwracana jest lista komunikatów (stringów)
 
-    public static List<String> validate(NewEventDto newEventDto) {
+    public static List<String> validate(NewEventDto newEventDto, TimeNow timeNow) {
         List<String> result = new ArrayList<>();
+        if (newEventDto == null) {
+            result.add("Event is null");
+            return result;
+        }
         if (newEventDto.getFromTime() == null) {
             result.add("From is null");
         }
-
         if (newEventDto.getToTime() == null) {
             result.add("To is null");
         }
         if (newEventDto.getFromTime() != null && newEventDto.getToTime() != null) {
             Duration duration = Duration.between(newEventDto.getFromTime(), newEventDto.getToTime());
-
             if (duration.isNegative()) {
                 result.add("To is before from");
             }
             if (duration.toMinutes() > 30L) {
-                result.add("Event is too long");
-            }
-            // - tutaj zaczyna się moja część
-            LocalDateTime now = LocalDateTime.now();
-            if (newEventDto.getFromTime().isBefore(now)) {
-                result.add("From is from the past");
+                result.add("Event too long");
             }
-            if (newEventDto.getFromTime().isBefore(now)) {
-                result.add("To is from the past");
+            if (newEventDto.getFromTime().isBefore(timeNow.now())) {
+                result.add("From is in the past");
             }
-            if (newEventDto.getFromTime().getHour() < 8 || newEventDto.getFromTime().getHour() > 16
-                    || newEventDto.getToTime().getHour() < 8 || newEventDto.getToTime().getHour() > 16) {
-                result.add("Selected hours are outside the opening hours (8:00 - 16:00)");
+            if (newEventDto.getToTime().isBefore(timeNow.now())) {
+                result.add("To is in the past");
             }
-            if (newEventDto.getItemName() == null) {
-                result.add("You didn't select the item");
+            if ((newEventDto.getFromTime().getHour() >= 16) || (newEventDto.getFromTime().getHour() <= 8)) {
```

Comment:
tutaj zrobiłabym tak:
```suggestion
            if ((newEventDto.getToTime().minusMinutes(1L).getHour() >= 16) || (newEventDto.getFromTime().getHour() < 8)) {
```